### PR TITLE
Containerization of the test environment for embedded systems

### PR DESCRIPTION
**Pull request by mateuszgrzelak**

### CODE REVIEW

File: blog/content/post/2020-12-14-docker-testing.md [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+of code with libraries, dependencies and other configuration and binaries needed
+to run it.
+
+Containers allow applications to run independently of the system they are
+running on. It is extremely useful when creating an application that runs on our
+computer, but not on another programmer's computer. It is possible that a
+program that uses many dependencies and libraries may not run on another
+computer due to incorrect versions. In this case, we only need to define
+necessary dependencies and pack our application in a container and regardless of
+the operating system, it should run on each computer with an installed Docker.
```
Comment (KBanecka) [inline blog/content/post/2020-12-14-docker-testing.md:39]: Containers allow applications to run independently of the system they are running on. It is handy when creating an application that runs on our computer but not on another programmer's computer. A program that uses many dependencies and libraries may not run on another computer due to incorrect versions. In this case, we only need to define necessary dependencies and pack our application in a container. Regardless of the operating system, it should run on each computer with an installed Docker.

File: blog/content/post/2020-12-14-docker-testing.md [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+whether data is correctly sent through the serial port. In our case, we write
+tests that anyone, regardless of the operating system, could run on their
+computer with Docker installed.
+
+To simulate our case, we use an Arduino UNO, which serves as the device
+under test (DUT), and Raspberry Pi, which tests our DUT. On the Arduino we
+check 4th and 8th pin. Unfortunately, the Raspberry can only read voltage not
+exceeding 3.3V. The range of GPIO pins of the Arduino is 0-5V so we have to
+use here a level shifter or voltage divider. In our case, we use six 1K
+resistors connected in the following way
```
Comment (KBanecka) [inline blog/content/post/2020-12-14-docker-testing.md:86]:
```suggestion
resistors connected in the following way:
```

File: blog/content/post/2020-12-14-docker-testing.md [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+
+![voltage_divider](/img/voltage_divider.jpg)
+
+![voltage_divider](/img/rpi_arduino_diagram.jpg)
+
+We need to know also how we can program Arduino. Usually, the Arduino IDE is
```
Comment (KBanecka) [inline blog/content/post/2020-12-14-docker-testing.md:92]:
```suggestion
We need to know how to program Arduino. Usually, the Arduino IDE is
```

File: blog/content/post/2020-12-14-docker-testing.md [hunk 4]
```diff
@@ -0,0 +1,375 @@
...
+depending on how many devices we have connected so this works only if we have
+one Arduino connected to RPi.
+
+To convert our code into the files that can be uploaded to Arduino we go to
+a specific folder and type `make`. This creates a new folder that contains all
+the necessary files. We can upload them by typing `make upload`. If we
+also want to remove this folder after uploading files to Arduino, we can simply
+add `make upload clean`.
+
+So we have all the necessary files for our DUT. It's time to write automatic
```
Comment (KBanecka) [inline blog/content/post/2020-12-14-docker-testing.md:187]:
```suggestion
Now we have all the necessary files for our DUT. It's time to write automatic
```

File: blog/content/post/2020-12-14-docker-testing.md [hunk 5]
```diff
@@ -0,0 +1,375 @@
...
+5 tests total, 5 passed, 0 failed
+==============================================================================
+Output:  /home/mateusz/blog_rf/output.xml
+Log:     /home/mateusz/blog_rf/log.html
+Report:  /home/mateusz/blog_rf/report.html
+```
+
+As we can see, the test results are the same except that now anyone can run our
+tests.
+
```
Comment (KBanecka) [inline blog/content/post/2020-12-14-docker-testing.md:367]: Thanks for getting here, If you have found Docker handy and worthy to use, check [this source](https://containers.3mdeb.com/) of docker containers maintained by our team or forked from Open Source projects with additional useful adjustments.

**Comment by mateuszgrzelak** [comment]
@KBanecka done

**Review by artur-rs — APPROVED** [review]

**Pull request merged**